### PR TITLE
link missing task or flow when merging configs

### PR DESCRIPTION
**Pull request by Br4nd0R**
# Changes
* Fixed an issue where CumulusCI was incorrectly merging task option overrides.

### QA Notes
I used the following in the standard lib while testing:
```
official_testing_flow:
        group: foo
        description: Does some stuff
        steps:
            1:
                task: util_sleep
            2:
                task: util_sleep
            3:
                task: util_sleep
   ```
   Then I modified the project `cumulusci.yml` to include:
   ```
   flows:
    official_testing_flow:
        steps:
            1:
                # task: util_sleep
                options:
                    seconds: 1
```
* Also verified setting the flow step to a value of `None` still works.

### CODE REVIEW

File: cumulusci/core/tests/test_utils_merge_config.py
Comment (prescod) [inline cumulusci/core/tests/test_utils_merge_config.py:267]: I'm not really a fan of this kind of testing. I think the automated tests should do what you described in the PR cmments. As a maintainer I don't care much that the new function `link_missing_task_or_flow` does the right thing. I care that CCI configs merge properly. I'd like to see some CCI configs merge. Then if I change the underlying implementation later, I'll still know that CCI configs merge properly.

For example, here is how we test the CCI validating linter.

https://github.com/SFDO-Tooling/CumulusCI/blob/8d46f48da6850d4b979d7f36f01f4b20103ed236/cumulusci/utils/yaml/tests/test_cumulusci_yml.py#L97

Comment (prescod) [reply cumulusci/core/tests/test_utils_merge_config.py:267]: IMO: A good rule of thumb is whether there's a clear line between the WI ACs and the test cases.

File: cumulusci/core/utils.py
Comment (davidmreed) [inline cumulusci/core/utils.py:260]: 👍

File: cumulusci/core/tests/test_utils_merge_config.py
Comment (davidmreed) [reply cumulusci/core/tests/test_utils_merge_config.py:267]: I agree that we should build new tests to catch this type of problem, and I will write a WI to do so (along with documenting this specific wrinkle of flow customization). However, I don't want to hold up @Br4nd0R's fix to this bug to do that, as there is serious production impact we need to cure.

**Review by davidmreed — APPROVED** [review]

**Pull request merged**
